examples/Custom Chain + Tool + Protect in Observe/Sample_Observe+Tool+Protect+NonLangchain.ipynb

```diff
@@ -0,0 +1,287 @@
+        "        prioritized_rulesets=[\n",
+        "            {\n",
+        "                \"rules\": [\n",
+        "                    {\n",
+        "                        \"metric\": \"pii\",\n",
+        "                        \"operator\": \"contains\",\n",
+        "                        \"target_value\": \"ssn\",\n",
+        "                    },\n",
+        "                ],\n",
+        "                \"action\": {\n",
+        "                    \"type\": \"OVERRIDE\",\n",
+        "                    \"choices\": [\n",
+        "                        \"Personal Identifiable Information detected in the model output. Sorry, I cannot answer that question.\"\n",
+        "                    ],\n",
+        "                },\n",
+        "            },\n",
+        "        ],\n",
+        "        stage_id=stage_id,\n",
+        "        timeout=3,\n",
+        "    )\n",
+        "    return protect_out"
+      ],
+      "metadata": {
+        "id": "UJgZwIf71Tc3"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# This is the function that we want the model to be able to call\n",
+        "def get_delivery_date(order_id: str) -> str:\n",
+        "    return datetime.datetime.now()"
+      ],
+      "metadata": {
+        "id": "BM6sg7ol88jW"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Define the tools you would like to use\n",
+        "tools = [\n",
+        "    {\n",
+        "        \"type\": \"function\",\n",
+        "        \"function\": {\n",
+        "            \"name\": \"get_delivery_date\",\n",
+        "            \"description\": \"Get the delivery date for a customer's order. Call this whenever you need to know the delivery date, for example when a customer asks 'Where is my package'\",\n",
+        "            \"parameters\": {\n",
+        "                \"type\": \"object\",\n",
+        "                \"properties\": {\n",
+        "                    \"order_id\": {\n",
+        "                        \"type\": \"string\",\n",
+        "                        \"description\": \"The customer's order ID.\",\n",
+        "                    },\n",
+        "                },\n",
+        "                \"required\": [\"order_id\"],\n",
+        "                \"additionalProperties\": False,\n",
+        "            },\n",
+        "        }\n",
+        "    }\n",
+        "]"
+      ],
+      "metadata": {
+        "id": "inqwSZsuwSXk"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Conversational Flow\n",
+        "inputs = [\n",
+        "{\"role\": \"system\", \"content\": \"You are a helpful customer support assistant. Use the supplied tools to assist the user.\"},\n",
```

Comment:
I would not include the system message as an input here. it's not an input from 'the user'. it's an instruction / your prompt template.